core/src/main/java/tc/oc/pgm/projectile/ProjectileModule.java

```diff
@@ -90,13 +96,53 @@ public ProjectileModule parse(MapFactory factory, Logger logger, Document doc)
             coolDown,
             throwable,
             precise,
-            blockMaterial);
+            blockMaterial,
+            scale,
+            solidBlockCollision,
+            maxTravelTime);
 
         factory.getFeatures().addFeature(projectileElement, projectileDefinition);
         projectiles.add(projectileDefinition);
       }
 
       return projectiles.isEmpty() ? null : new ProjectileModule(ImmutableSet.copyOf(projectiles));
     }
+
+    private static final Map<String, ProjectileDefinition.ProjectileEntity.AbstractEntity> ABSTRACT_ENTITY_MAP =
+        ImmutableMap.of(
+          "block", new ProjectileDefinition.ProjectileEntity.AbstractEntity(
+                  ProjectileDefinition.ProjectileEntity.AbstractEntityType.BLOCK
+              )
+        );
+
+    private static ProjectileDefinition.ProjectileEntity parseProjectileEntity(
+      final Element element,
+      final String attributeName,
+      final Class<? extends Entity> def
+    ) throws InvalidXMLException {
+      final Node node = Node.fromAttr(element, attributeName);
+      if (node == null) {
+        return new ProjectileDefinition.ProjectileEntity.RealEntity(def);
+      }
+      final String entityText = node.getValue();
+      if (!entityText.matches("[a-zA-Z0-9_]+")) {
+        throw new InvalidXMLException("Invalid entity type '" + entityText + "'", node);
+      }
+      if (ABSTRACT_ENTITY_MAP.containsKey(entityText.toLowerCase())) {
+        return ABSTRACT_ENTITY_MAP.get(entityText.toLowerCase());
+      }
+      return new ProjectileDefinition.ProjectileEntity.RealEntity(XMLUtils.parseEntityTypeAttribute(element, attributeName, def));
```

Comment:
Given custom logic will likely have to be involved, consider just using a switch, and using the fluent parser:

```suggestion
      return switch(entityText.toLowerCase(Locale.ROOT)) {
        case "entity" -> new ProjectileDefinition.BlockEntity(
                parser.parseFloat(el, "scale").optional(1.0f),
                parser.parseBool(el, "solid-block-collision").orTrue(),
                parser.parseDuration(el, "max-travel-time").optional(DEFAULT_MAX_TRAVEL);
        default -> new ProjectileDefinition.RealEntity(XMLUtils.parseEntityTypeAttribute(node));
      }
```